src/asInput/index.jsx

```diff
@@ -31,7 +32,7 @@ const asInput = (WrappedComponent, labelFirst = true) => {
       this.handleChange = this.handleChange.bind(this);
       this.handleBlur = this.handleBlur.bind(this);
 
-      const id = newId('asInput');
+      const id = this.props.id ? this.props.id : newId('asInput');
```

Comment:
would `this.props.id || newId('asInput');` be slightly more readable?

Reply:
Sure I can make that swap - I tend to lean ternary as ternaries are what I assume to be more familiar across languages, but I know we use the `||` syntax elsewhere for react so it won't be a stand out.

Note -- just saw your second comment, see reply there as that might make this unnecessary.